### ISSUE TITLE
value-keyword-case should ignore custom-idents (like font names and animation names)

### ISSUE THREAD

**Issue by evilebottnawi** [stylelint/stylelint#1101]
- [x] animation
- [x] animation-name
- [x] font
- [x] font-family
- [x] counter-increment
- [x] grid-row
- [x] grid-column
- [x] grid-area
- [x] list-style
- [x] list-style-type

> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?
```css
h1 {
    font-family: Geneva, Arial, Helvetica, sans-serif;
}
```
Also ignore in short `font` property

**Comment by jeddy3** [comment]
Ref: https://github.com/stylelint/stylelint/issues/1082

According to [the specs](https://drafts.csswg.org/css-values/#custom-idents) properties that accept *custom-idents* must do in a "positionally unambiguous" way. Therefore, I think, we could identify which properties accept *custom-idents* and in which value position, and then *only* ignore keywords found in those positions.

Off hand, I can think of the following properties that accept *custom-idents*:

- `animation`
- `animation-timing-function`
- `animation-name`
- `will-change`
- `font`
- `font-family`
- `counter-increment`
- `grid-row`
- `grid-column`
- `grid-area`
- `list-style-type`

Does this seem like a valid approach?

**Comment by evilebottnawi** [comment]
@jeddy3 https://github.com/stylelint/stylelint/issues/1089 laso ignored, maybe close - duplicate

**Comment by jeddy3** [comment]
Nope, I think that's a different issue. That's specific to ignoring SVG keywords associated with the SVG `text-rendering` property. I don't think it's related to *custom-idents* and should be solved slightly differently e.g. [like how treat](https://github.com/stylelint/stylelint/blob/master/src/rules/value-keyword-case/index.js#L46) `currentColor`.

**Comment by davidtheclark** [comment]
Seems valid to me.

**Comment by jeddy3** [comment]
@evilebottnawi Any change of prioritising this issue over adding new rules as it affect `stylelint-config-standard`?

**Comment by evilebottnawi** [comment]
@jeddy3 today, tomorrow i PR

**Comment by alienlebarge** [comment]
What if values are set in a variables ?

```css
:root {
    --base-font-family: Geneva, Arial, Helvetica, sans-serif;
}

h1 {
    font-family: var(--base-font-family);
}
```

**Comment by davidtheclark** [comment]
Since there would be no way to know what a variable is intended for, we either have to ignore all variables completely or leave it to users to use comments to disable when needed. Which sounds for user-friendly to you, @alienlebarge ?

**Comment by alienlebarge** [comment]
I see what you mean.
In my case, adding comments to disable will be more useful.

But do you think it's possible (and a good thing) to select what to ignore ?

```json
{
    rules: {
        ...
         "at-rule-empty-line-before": [
            "lower", {
                except: ["variables", "ignore-comments"]
            }
        ],
        ...
    }
}
```

Another thing I see is that if we choose to ignore variables, we have to enable it for all kinds of variables (CSS Variables, Sass, LESS, *future-preprocessors*, ...)

```sass
/* CSS Variables*/
:root {
    --base-font-family: Geneva, Arial, Helvetica, sans-serif;
}

/* Sass Variables */
$base-headings-family: Futura, "Trebuchet MS", Arial, sans-serif;

h1 {
    font-family: var(--base-headings-family);
}

p {
    font-family: $base-headings-family;
}
```

**Comment by evilebottnawi** [comment]
@jeddy3 i can't found custom idents for `animation-timing-function`

**Comment by jeddy3** [comment]
> @jeddy3 i can't found custom idents for animation-timing-function

Yep, you're right. Lets knock `animation-timing-function` off the list.

**Comment by evilebottnawi** [comment]
@jeddy3 I can not find an example of customs idents with `will: change`.

**Comment by jeddy3** [comment]
https://developer.mozilla.org/en/docs/Web/CSS/will-change ?

**Comment by evilebottnawi** [comment]
> <custom-ident>
Indicates that the author expects to animate or change the property with the given name on the element in the near future.

All properties valid and lower and upper.

**Comment by jeddy3** [comment]
Actually, you're right. Lets skip `will-change`.

**Comment by jeddy3** [comment]
@evilebottnawi Is there anything left to do on this, or can we close?

**Comment by evilebottnawi** [comment]
@jeddy3 i am working on this issue, tomorrow i PR and we can close issue.

**Comment by evilebottnawi** [comment]
@jeddy3 last util `findAnimationName`

**Comment by jeddy3** [comment]
> i am working on this issue, tomorrow i PR and we can close issue.

Excellent.

**Comment by evilebottnawi** [comment]
https://github.com/stylelint/stylelint/pull/1445

**Comment by evilebottnawi** [comment]
Close https://github.com/stylelint/stylelint/pull/1445
